auth/client-api.md

```diff
@@ -0,0 +1,24 @@
+# Client API
+
+```typescript
+abstract class Client {
+  // ---------- Methods ----------------------------------------------- //
+
+  // initializes the client with persisted storage and a network connection
+  public abstract init(): Promise<void>;
+
+  // for responder to pair a pairing created by a proposer
+  public abstract pair(params: {
+    uri: string;
+  }): Promise<Sequence>;
+
+  // request wallet authentication
+  public abstract request(params: RequestParams): Promise<boolean>;
```

Comment:
```suggestion
  // request wallet authentication
  // returns URI if pairing topic not provided
  public abstract request(params: RequestParams, pairingTopic?: string): Promise<string?>;
```

Reply:
I mean... this is a weird one to spec out because the Sign API also doesn't have this specced out

However the Javascript pattern is two have either a URI as an output or not and additionally a Promise to be resolved

I added a pairing_created to the spec like Sign API for consistency